opentelemetry/src/sdk/trace/span_processor.rs

```diff
@@ -71,10 +76,10 @@ pub trait SpanProcessor: Send + Sync + std::fmt::Debug {
     /// API, therefore it should not block or throw an exception.
     fn on_end(&self, span: SpanData);
     /// Force the spans lying in the cache to be exported.
-    fn force_flush(&self);
+    fn force_flush(&self) -> ExportResult;
     /// Shuts down the processor. Called when SDK is shut down. This is an
     /// opportunity for processors to do any cleanup required.
-    fn shutdown(&mut self);
+    fn shutdown(&mut self) -> ExportResult;
```

Comment:
I think I would prefer to use a different result type here. The spec mentions `ExportResult` as something the export function returns. The functions `force_flush` and `shutdown` don't have a defined result type in the spec. So maybe we just define one ourselves. I think `Result<(), Box<dyn Error>>` or even `TraceError` would be fine.

Reply:
Yeah `ExportResult` is actually an alias for `Result<(), Box<dyn std::error::Error + Send + Sync + 'static>>`. Will update to make this clear.